src/lib/operators/projection.cpp

```diff
@@ -116,23 +127,27 @@ const std::string Projection::_get_type_of_expression(const std::shared_ptr<Expr
   const auto type_left = _get_type_of_expression(expression->left_child(), table);
   const auto type_right = _get_type_of_expression(expression->right_child(), table);
 
+  if (type_left == "NULL") return type_right;
+  if (type_right == "NULL") return type_left;
+
   // TODO(anybody): int + float = float etc...
   // This is currently not supported by `_evaluate_expression()` because it is only templated once.
-  Assert(type_left == type_right, "Projection currently only supports expressions with same type on both sides.");
+  Assert(type_left == type_right, "Projection currently only supports expressions with same type on both sides (" +
+                                      type_left + " vs " + type_right + ")");
   return type_left;
 }
 
 template <typename T>
-const tbb::concurrent_vector<T> Projection::_evaluate_expression(const std::shared_ptr<Expression>& expression,
-                                                                 const std::shared_ptr<const Table> table,
-                                                                 const ChunkID chunk_id) {
+const pmr_concurrent_vector<std::optional<T>> Projection::_evaluate_expression(
+    const std::shared_ptr<Expression>& expression, const std::shared_ptr<const Table> table, const ChunkID chunk_id) {
```

Comment:
I am not 100% happy with having vectors of optionals here. This wastes a byte per row. However, for compatibility with `std::plus` and so on, this is really handy. Also, these vectors are only used during execution and will then be destroyed.